extras/wait-for-bricks.sh

```diff
@@ -92,34 +48,36 @@ function volumes_waiting_for_bricks_up()
         local vol_down_info
         while read -r volname
         do
-                if [[ "Started" == "$(volinfo_field "$volname" 'statusStr')" ]]
+                read -r total_brick_count
+                brick_count="$(volume_online_brick_count "$volname" "$total_brick_count")"
+                if [[ $brick_count -ne $total_brick_count ]]
                 then
-                        if [[ "Y" == "$(is_enabled "$(volinfo_option "$volname" features.shard)")" ]]
+                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
+                        if [[ -z "$wait_info" ]]
                         then
-                                brick_count="$(volume_online_brick_count "$volname")"
-                                total_brick_count=$(volinfo_field "$volname" 'brickCount')
-                                if [[ $brick_count -ne $total_brick_count ]]
-                                then
-                                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
-                                        if [[ -z "$wait_info" ]]
-                                        then
-                                                wait_info="$vol_down_info"
-                                        else
-                                                wait_info="$wait_info, $vol_down_info"
-                                        fi
-                                fi
+                                wait_info="$vol_down_info"
+                        else
+                                wait_info="$wait_info, $vol_down_info"
                         fi
                 fi
-        done < <(gluster volume list)
+        done < <(block_volnames_brick_count_get)
         if [[ ! -z "$wait_info" ]]; then echo "$wait_info"; fi
 }
 
 function check_glusterd_running()
 {
-        if ! pidof glusterd > /dev/null 2>&1
+        if ! command -v ps > /dev/null 2>&1
```

Comment:
ps is part of package procps-ng, and should be defaultly shipped at all OS flavors.
IMO this check is not needed (just an opinion), and that should not block this patch :-)

Reply:
In containers, quite a few packages keep missing, for example, I have never seen an OS installtion without less, but I have seen quite a few containers without 'less' command. I don't think it hurts to keep this code. I don't mind adding the same checks for sed, grep, awk, head, echo as well.

Reply:
Let me also restructure that part of the code.